File: poo/3. Newton's Laws Simulator  JS/action.java
public class Action extends Forces {
    private final int action;
    private int reaction;

    public Action(int action, int reaction) {
        super("Action and Reaction", "Acción y Reacción");
        this.action = action;
        this.reaction = reaction;
    }

    public int getAction() {
        return action;
    }

    public int getReaction() {
        return reaction;
    }

    public Float calculateForce() {
        this.reaction = -this.action;
        return (float) reaction;
    }
}

File: poo/3. Newton's Laws Simulator/action.php
<?php 
    class action extends forces {
        readonly private int $action;
        private int $reaction;

        public function __construct(int $action, int $reaction) {
            parent::__construct("Action and Reaction", "Acción y Reacción");
            $this->action = $action;
            $this->reaction = $reaction;
        }
        public function getAction(): int {
            return $this->action;
        }
        public function getReaction(): int {
            return $this->reaction;
        }
        public function calculateForce(): ?float {
            $this->reaction = - $this->action;
            return $this->reaction;
        }
    }
?>